app/src/main/java/in/co/rajkumaar/amritarepo/timetable/AcademicTimetableActivity.java

```diff
@@ -1,303 +1,341 @@
+        branch.setAdapter(courseAdapter);
+        if (course.getSelectedItemPosition() == pref.getInt("course", 0)) {
+            branch.setSelection(pref.getInt("branch", 0));
+        }
+    }
+
+
+    private void loadLists() {
+        courses.add("[Choose course]");
+        if (!pref.contains("courses")) {
+            getCourse();
+
+        } else {
+            courses = new ArrayList<>();
+            Gson gson = new Gson();
+            String json = pref.getString("courses", null);
+            Type listType = new TypeToken<ArrayList<String>>() {
+            }.getType();
+            ArrayList<String> course = gson.fromJson(json, listType);
+            courses.addAll(course);
+        }
+        ArrayAdapter<String> courseAdapter = new ArrayAdapter<>(AcademicTimetableActivity.this, R.layout.spinner_item1, courses);
+        courseAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        course.setAdapter(courseAdapter);
+
+        sems = new ArrayList<>();
+        sems.add("[Choose semester]");
+        int i = 1;
+        while (i <= 10) {
+            sems.add(String.valueOf(i++));
+        }
+        ArrayAdapter<String> semAdapter = new ArrayAdapter<>(this, R.layout.spinner_item1, sems);
+        semAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        sem.setAdapter(semAdapter);
+
+        batches.add("[Choose batch]");
+        batches.add("A");
+        batches.add("B");
+        batches.add("C");
+        batches.add("D");
+        batches.add("E");
+        batches.add("F");
+        ArrayAdapter<String> batchAdapter = new ArrayAdapter<>(this, R.layout.spinner_item1, batches);
+        batchAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        batch.setAdapter(batchAdapter);
+
+        ArrayAdapter<String> yearAdapter = new ArrayAdapter<>(this, R.layout.spinner_item1, Utils.getAcademicYears());
+        yearAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        year.setAdapter(yearAdapter);
+
+    }
+
+    private void getCourse() {
+        AsyncHttpClient client = new AsyncHttpClient();
+        client.get(getString(R.string.course_name_url), new AsyncHttpResponseHandler() {
+            @Override
+            public void onSuccess(int i, Header[] headers, byte[] bytes) {
+                Document html = Jsoup.parse(new String(bytes));
+                Elements rows = html.select("center > table>tbody>tr").select("td>table>tbody>tr>td>form>select");
+                for (Element row : rows) {
+                    Elements columns = row.getElementsByTag("option");
+                    if (columns.get(0).text().equals("Course")) {
+                        for (int j = 1; j < columns.size(); j++) {
+                            String parsedCourse = columns.get(j).attr("value");
+                            courses.add(parsedCourse);
+                        }
```

Comment:
Can you add a `break` statement here ?

Reply:
Changed to getElementById("#id") and removed the for loop